### PR TITLE
Update common resources dependency

### PR DESCRIPTION
**Pull request by youfoundron**
Replaces deprecated dependency `ethereum-common`  with `ethereumjs-common`.

### CODE REVIEW

**Comment by youfoundron** [comment]
@holgerd77 

Lol I'm not sure how I didn't catch that there was a whole API in the docs, thought I this was basically a repository of config files.

I've updated PR to use params, will look into the block library shortly.

**Comment by holgerd77** [comment]
Hi @youfoundron, this already looks much better, thanks for the update! :-)

Actually we had various updates along this line in the [block](https://github.com/ethereumjs/ethereumjs-block/pull/44) but also in the blockchain and VM repos adding network and hardfork support to the libraries by expanding the external constructor API (for both normal ``index.js`` and fake ``fake.js`` transaction) and allowing to pass either ``chain`` and ``hardfork`` parameter in a separate additional ``opts``dict in the constructor (for explicitly fixing both parameters) or alternatively having a ``common`` instance passed (e.g. for implicitly taking the values from a block).

I think this would be the way to go here as well, would you be willing to go along this extra mile? Would be cool! 😄

For the API this should be very analogue to the [code](https://github.com/ethereumjs/ethereumjs-block/blob/master/index.js#L23) (see also the JSDoc comments along) in the block library.

In the past we have always somewhat implicitly updated our libraries to the latest hardfork rules. So for convenience and to remain compatibility hardfork should probably also default to ``byzantium`` for now like being done [here](https://github.com/ethereumjs/ethereumjs-vm/blob/master/lib/index.js#L48) in the VM.

Main indicator that a post-chain start version of tx is taken as default in the library (I am also not so totally deep into the tx library) is that the current ``homestead`` flag is set to ``true`` by [default](https://github.com/ethereumjs/ethereumjs-tx/blob/master/index.js#L136). This whole flag should be removed, since this is replaced in a more general way through the hardfork setting. All other references to the flag should be replaced with ``_common.gteHardfork('...')`` comparisons similar to [here](https://github.com/ethereumjs/ethereumjs-vm/blob/db3db1c333e2f8ced7d14eec28a2f0ccc0318c7a/lib/opFns.js#L161) in the VM (so logic from the method is "if the hardfork set in ``common`` is greater or equal than the hardfork stated do x"). Probably it is safe to use ``homestead`` here, maybe we should nevertheless have a closer look if it was really homestead which introduced the respective changes (e.g. I don't see any new gas price changes in the [homestead.json](https://github.com/ethereumjs/ethereumjs-common/blob/master/hardforks/homestead.json) file in the common library. This might very well also be an inconsistency/bug in the common library though).

I've seen that you already gave some thumbs up on the comment on the tests PR from @danjm, would be great if you guys could coordinate on that, than Dan can directly use the result from here for his test PR.

**Comment by holgerd77** [comment]
Thanks for this, this looks already really good! 😄 Might take some days till review.

File: test/transactionRunner.js [hunk 1]
```diff
@@ -33,9 +32,6 @@ testing.runTests(function (testData, sst, cb) {
   try {
     var rawTx = ethUtil.toBuffer(testData.rlp)
     var tx = new Tx(rawTx)
-    if (testData.blocknumber !== String(common.homeSteadForkNumber.v)) {
```
Comment (danjm) [inline test/transactionRunner.js]: To keep this consistent with the changes to index.js, `if (testData.blocknumber < String(common.homeSteadForkNumber.v)) {` would be preferable here. However, as my PR to rewrite the `transactionRunner.js` file will be merged right after this PR, I think this change is okay.

**Comment by danjm** [comment]
@youfoundron this looks good! I have left some comments for some small changes.

@holgerd77 I have reviewed this as I have been further refining the test rewrite PR. My latest commit to that PR is compatible with the changes introduced here, and can be rebased once this is merged.

I merged both PRs locally and tested them along with further changes we need for "Spurious Dragon" compatibility (for which I will make another PR). Everything seems to be working together smoothly.

**Comment by youfoundron** [comment]
@holgerd77 @danjm Great feedback. Will review this weekend.

**Comment by holgerd77** [comment]
Hi @youfoundron, thanks for the great work so far! 😄 This PR is currently blocking other work, would it be ok for you if @danjm re-pushes your state of the work directly to the repository and does the latest changes required?

File: index.js [hunk 2]
```diff
@@ -41,10 +41,33 @@ const N_DIV_2 = new BN('7fffffffffffffffffffffffffffffff5d576e7357a4501ddfe92f46
...
+    if (opts.common) {
+      if (opts.chain) {
+        throw new Error('Instantiation with both opts.common and opts.chain parameter not allowed!')
+      }
+      this._common = opts.common
+    } else {
+      let chain = opts.chain ? opts.chain : 'mainnet'
+      let hardfork = opts.hardfork ? opts.hardfork : 'byzantium'
+      let supportedHardforks = [
+        'byzantium',
```
[the remark this answers is not in the document]
Comment (youfoundron) [reply index.js]: I've removed the `supportedHardforks` setting.

File: index.js [hunk 3]
```diff
@@ -133,7 +156,7 @@ class Transaction {
 
     // set chainId
     this._chainId = chainId || data.chainId || 0
-    this._homestead = true
+    this._homestead = this._common.gteHardfork('homestead')
```
[the remark this answers is not in the document]
Comment (youfoundron) [reply index.js]: I've replaced all `this._homestead` references with `this._common.gteHardfork('homestead')`

**Comment by youfoundron** [comment]
I've made the requested changes. Any idea what is up with tests failing? Not certain what to make of logs in CircleCI.

**Comment by holgerd77** [comment]
Great, thanks for the updates!

Hmm, not sure if this was simply some Travis malfunctioning, since on the job run I checked output just stopped at some point. Have retriggered the failing jobs, sometimes these pass on a second run (and it can be relatively clearly attributed to Travis failing and not the tests).

File: test/transactionRunner.js (hunk 1 above)
Comment (holgerd77) [reply test/transactionRunner.js]: It's very likely that tests are failing here cause of the removal of this comparison and therefore have changed conditions.

I would just do a hard-coded comparison here and then instantiate the tx with the corresponding HF in the options, so:

```javascript
if (testData.blockNumber >== 1150000) {
  // instantiate a Homestead+ tx
} else {
  // instantiate a pre-homestead (chainstart) tx
}
```

Then we recreate the former conditions (you might even want to just use the same `!==` comparison) and it should be easier to get this passed for now. Dan can then do subsequent test optimization work in his own PR. Does this make sense?

Comment (youfoundron) [reply test/transactionRunner.js]: Excellent will give this a shot after work.

**Pull request merged**